github/commands/open_on_remote.py

```diff
@@ -13,17 +14,60 @@ class GsOpenFileOnRemoteCommand(TextCommand, GitCommand, git_mixins.GithubRemote
     """
     Open a new browser window to the web-version of the currently opened
     (or specified) file. If `preselect` is `True`, include the selected
-    lines in the request.
+    lines in the request. If the active tracked remote is the same as the
+    integrated remote, open browser directly, if not, display a list to remotes
+    to choose from.
 
     At present, this only supports github.com and GitHub enterprise.
     """
 
-    def run(self, edit, preselect=False, fpath=None):
-        fpath = fpath or self.get_rel_path()
+    def run(self, edit, remote=None, preselect=False, fpath=None):
+        sublime.set_timeout_async(
+            lambda: self.run_async(edit, remote, preselect, fpath))
```

Comment:
No need to pass on `edit` here.  In fact, for future reference, that `edit` object is unusable once `run` has completed execution.  Any edits need to happen synchronously, which is why we trigger an event whenever we want a content change to occur.